base/mpfr.jl

```diff
@@ -276,9 +274,9 @@ promote_rule(::Type{BigFloat}, ::Type{<:AbstractFloat}) = BigFloat
 big(::Type{<:AbstractFloat}) = BigFloat
 
 function convert(::Type{Rational{BigInt}}, x::AbstractFloat)
-    if isnan(x); return zero(BigInt)//zero(BigInt); end
-    if isinf(x); return copysign(one(BigInt),x)//zero(BigInt); end
-    if x == 0;   return zero(BigInt) // one(BigInt); end
+    isnan(x) && return zero(BigInt) // zero(BigInt)
+    isinf(x) && return copysign(one(BigInt),x) // zero(BigInt)
+    x == 0 && return zero(BigInt) // one(BigInt)
```

Comment:
As you already changed such a test to `iszero` earlier, and you are touching this function, you might as well use `iszero` here too?